metric/registry.go

```diff
@@ -70,6 +70,38 @@ func (r *Registry) AddInt64Gauge(name, description string, unit metricdata.Unit,
 	return i, nil
 }
 
+// AddInt64DerivedGauge creates and adds a new derived int64-valued gauge to this registry.
+// A derived gauge is convenient form of gauge where the object associated with the gauge
+// provides its value by implementing func() int64.
+func (r *Registry) AddInt64DerivedGauge(name, description string, unit metricdata.Unit, labelKeys ...string) (*Int64DerivedGauge, error) {
```

Comment:
Before we make everything to work I would suggest to modify this function:

func (r *Registry) AddInt64DerivedGauge(name, description string, unit metricdata.Unit, constantLabels map[LabelKey]LabelValue, labelKeys ...string)

For sure in a different PR but I just remember that.

Reply:
sure. I have another PR coming up for #1063. I can update along with that..